src/main/java/org/junit/runners/ParentRunner.java

```diff
@@ -209,9 +209,8 @@ private boolean areAllChildrenIgnored() {
      * Returns a {@link Statement}: run all non-overridden {@code @BeforeClass} methods on this class
      * and superclasses before executing {@code statement}; if any throws an
      * Exception, stop execution and pass the exception on.
-     * @param testClass 
      */
-    public static Statement withBeforeClasses(Statement statement, TestClass testClass) {
+    protected Statement withBeforeClasses(Statement statement) {
```

Comment:
@PeterWippermann I don't understand what you mean by "I can't call ParentRunner.classBlock(...) in my custom Runner because between invoking the children and applying the class statements, I'd like to apply @TestRules". The method `withClassRules()` applies all of the rules annotated with `@ClassRule`. You shouldn't invoke rules annotated with `TestRule` in `ParentRunner` because there is no instance of the class.